src/apps/intermediate-list/stackable-fees/fee-details-content.tsx

```diff
@@ -0,0 +1,130 @@
+import dayjs from "dayjs";
+import * as React from "react";
+import { FC, useState } from "react";
+import { Link } from "../../../components/atoms/link";
+import CheckBox from "../../../components/checkbox/Checkbox";
+import { FeeV2 } from "../../../core/fbs/model";
+import { useText } from "../../../core/utils/text";
+import { FaustId } from "../../../core/utils/types/ids";
+import { useUrls } from "../../../core/utils/url";
+import FeeStatusCircle from "../utils/fee-status-circle";
+import { isDateBeforePaymentChangeDate } from "../utils/intermediate-list-helper";
+import StackableFeesList from "./stackable-fees-list";
+
+export interface FeeDetailsContentProps {
+  feeDetailsData: FeeV2;
+}
+
+const FeeDetailsContent: FC<FeeDetailsContentProps> = ({ feeDetailsData }) => {
+  const t = useText();
+  const [check, setCheck] = useState(false);
+  const { termsOfTradeUrl } = useUrls();
+  const handleAcceptedTerms = () => {
+    setCheck(!check);
+  };
+  const {
+    amount = 0,
+    creationDate = "",
+    dueDate = "",
+    materials = []
+  } = feeDetailsData;
+  const prePaymentTypeChange = isDateBeforePaymentChangeDate(new Date(dueDate));
+  const creationDateFormatted = dayjs(creationDate).format("D. MMMM YYYY");
+
+  return (
+    <div className="modal modal-show modal-loan">
+      <div className="modal__screen-reader-description" id="describemodal">
+        {t("feeDetailsModalScreenReaderText")}
+      </div>
+      <div className="modal-loan__container">
+        <div className="modal-loan__header">
+          <div className="mr-32">
+            <FeeStatusCircle
+              dueDate={dueDate || ""}
+              feeCreationDate={creationDate}
+            />
+          </div>
+          <div>
+            <h2 className="modal-loan__title text-header-h2">
+              {t("turnedInText", {
+                placeholders: { "@date": creationDateFormatted }
+              })}
+            </h2>
+          </div>
+        </div>
+        <div className="modal-loan__buttons">
+          <CheckBox
+            id="checkbox_id__fee_details"
+            onChecked={() => handleAcceptedTerms()}
+            label={
+              <>
+                {t("iAcceptText")}{" "}
+                <Link href={new URL(termsOfTradeUrl)}>
+                  {t("termsOfTradeText")}
+                  <sup>*</sup>
+                </Link>
+              </>
+            }
```

Comment:
I see some code that is repeated in `TotalPaymentPay` is it not possible to make a subcomponent of one or multiple blocks here and reuse that?